### PR TITLE
Export template function

### PR DESCRIPTION
**Pull request by GeorgeTaveras1231**
This facilitates wrapping template behavior in other functions, and not needing to reference the tag function in many places.

In my case, I've created custom `Console` class that can be configured to have different prefixes, I'd like to be able to instantiate the class with a color template without needing to import `chalk` or `chalk-template` in multiple places

```js
const scopedConsole = new Console(/*Prefix*/'{blue SomePrefix}')
```

### CODE REVIEW

**Comment by sindresorhus** [comment]
I'm indifferent to exposing this. Deferring to @Qix-. However, this will need tests, docs, and TS types for it to be mergable.

**Comment by Qix-** [comment]
Exposing it is fine, but agreed, needs tests, types and docs.

Docs need to mention that all input needs to be properly escaped - a lot of the "safety" comes from using tagged literals.

**Comment by GeorgeTaveras1231** [comment]
@Qix- @sindresorhus I've updated the branch to include tests, types, and docs. I was a little unsure about what to do but welcome any feedback. I also added typescript as a dev dependency because it seemed that `npm test` requires typescript.

**Comment by Qix-** [comment]
Thank you :) Published as `0.4.0`

File: index.d.ts [hunk 1]
```diff
@@ -21,3 +21,15 @@ log(chalk.red.bgBlack(chalkTemplate`2 + 3 = {bold ${2 + 3}}`));
...
 export default function chalkTemplate(text: TemplateStringsArray, ...placeholders: unknown[]): string;
+
+/**
+Terminal string styling. It is preferred that you use the template tag (default export) but this function is useful if you'd like to wrap the color template function.
+
+@example
+```
+import { template } from 'chalk-template';
+
+log(template('Today is {red hot}')
```
Comment (sindresorhus) [inline index.d.ts:32]: Multiple typos

File: index.d.ts [hunk 2]
```diff
@@ -21,3 +21,15 @@ log(chalk.red.bgBlack(chalkTemplate`2 + 3 = {bold ${2 + 3}}`));
 ```
 */
 export default function chalkTemplate(text: TemplateStringsArray, ...placeholders: unknown[]): string;
+
+/**
+Terminal string styling. It is preferred that you use the template tag (default export) but this function is useful if you'd like to wrap the color template function.
+
+@example
+```
+import { template } from 'chalk-template';
```
Comment (sindresorhus) [inline index.d.ts:30]: Does not follow the existing code style.

File: package.json [hunk 3]
```diff
@@ -49,6 +49,7 @@
 		"ava": "^3.15.0",
 		"cross-env": "^7.0.3",
 		"tsd": "^0.18.0",
+		"typescript": "^4.6.2",
```
Comment (sindresorhus) [inline package.json:52]: Don't make unrelated changes.

File: readme.md [hunk 4]
```diff
@@ -58,6 +58,16 @@ Note that function styles (`rgb()`, etc.) may not contain spaces between paramet
 
 All interpolated values (`` chalkTemplate`${foo}` ``) are converted to strings via the `.toString()` method. All curly braces (`{` and `}`) in interpolated value strings are escaped.
 
+## Template function
+
```
Comment (sindresorhus) [inline readme.md:62]: Readme is not in sync with index.d.ts

**Comment by sindresorhus** [comment]
> Docs need to mention that all input needs to be properly escaped - a lot of the "safety" comes from using tagged literals.

This was not done.

**Comment by sindresorhus** [comment]
I cleaned this up in https://github.com/chalk/chalk-template/commit/bf830d7b965c8c6144fbe0793ef4cc89480435d9.

**Comment by Qix-** [comment]
Whoops sorry @sindresorhus, missed the mark on this one. Thanks for cleaning it up.
